bin/tx-prover/src/proxy/worker.rs

```diff
@@ -0,0 +1,102 @@
+// WORKER
+// ================================================================================================
+
+use std::time::Duration;
+
+use pingora::lb::Backend;
+use tonic::transport::Channel;
+use tonic_health::pb::{
+    health_check_response::ServingStatus, health_client::HealthClient, HealthCheckRequest,
+};
+use tracing::error;
+
+use crate::commands::WorkerConfig;
+
+/// A worker used for processing of requests.
+///
+/// A worker consists of a backend service (defined by worker address), a flag indicating wheter
+/// the worker is currently available to process new requests, and a gRPC health check client.
+#[derive(Debug, Clone)]
+pub struct Worker {
+    pub(crate) worker: Backend,
+    health_check_client: HealthClient<Channel>,
+    pub(crate) is_available: bool,
```

Comment:
Why do some fields here need to be `pub(crate)`? We already have `Self::is_available()` method and we could probably do the same for the `worker` field (I'd probably rename it into `backend`).